agent/acs/handler/attach_eni_handler_common.go

```diff
@@ -42,10 +44,26 @@ func (handler *ackTimeoutHandler) handle() {
 	}
 	if !eniAttachment.IsSent() {
 		seelog.Warnf("Timed out waiting for ENI ack; removing ENI attachment record with MAC address: %s", handler.mac)
+		handler.removeENIAttachment(handler.mac)
 		handler.state.RemoveENIAttachment(handler.mac)
 	}
 }
 
+func (handler *ackTimeoutHandler) removeENIAttachment(mac string) {
+	attachmentToRemove, ok := handler.state.ENIByMac(mac)
+	if ok {
```

Comment:
Should it throw an error when it's not `ok`

Reply:
Done.

Reply:
`If ok` not required here